src/server/io_utils.cc

```diff
@@ -16,14 +16,22 @@ io::Result<size_t> BufferedStreamerBase::WriteSome(const iovec* vec, uint32_t le
   return io::BufSink{&producer_buf_}.WriteSome(vec, len);
 }
 
-void BufferedStreamerBase::NotifyWritten() {
+void BufferedStreamerBase::NotifyWritten(bool allow_await) {
   if (IsStopped())
     return;
   buffered_++;
   // Wake up the consumer.
   waker_.notify();
   // Block if we're stalled because the consumer is not keeping up.
-  waker_.await([this]() { return !IsStalled() || IsStopped(); });
+  waker_.await([this, allow_await]() { return !allow_await || !IsStalled() || IsStopped(); });
+}
+
+void BufferedStreamerBase::WakeIfWritten() {
+  if (IsStopped())
+    return;
+  if (buffered_) {
+    waker_.await([this]() { return !IsStalled() || IsStopped(); });
+  }
```

Comment:
We wake by using `waker_.notify()`, the only thing that this can do is blocking

Reply:
Yes I will change the function name, we already notiflied the consumer, we just need to block now till consumer runs